### ISSUE TITLE
Web3 methods promise stuck on pending and doesn't return any value using ganache provider with React.js

### ISSUE THREAD

**Issue by jmarrr** [ConsenSys-archive/ganache#724]
I tried to reproduce my issue on a new fresh react project and used ganache core to fork mainnet and still persist the problem where the methods like `getBlockNumber`, `getGasPrice`, etc doesn't return anything.

## Expected Behavior
 `getBlockNumber`, `getGasPrice` or other methods should return their respective value. The `console.log` in the snippet below should return a value.

## Current Behavior
 `getBlockNumber`, `getGasPrice` or other methods doesn't return any value or stuck at pending promise. I used await.

## Steps to Reproduce (for bugs)
1. Create a fresh react project. 
2. Install web3, ganache-core.
3. Update the whole `App.js` with the script below.
4. `npm start` and check the console.

Snippet to reproduce the issue:
```
// App.js
import React from "react";
import logo from './logo.svg';
import './App.css';
const Web3 = require('web3');
const ganache = require("ganache-core");
const web3ws = new Web3(new Web3.providers.WebsocketProvider('ws://localhost:8545'));

class App extends React.Component {
  async componentDidMount() {
    document.title = 'Sample App';
    const checker = await this.txChecker(); 
  }

  checker = async () => {
    let latest_block = await web3ws.eth.getBlockNumber()
    console.log("latest_block main: ", latest_block)
    const localWeb3 = new Web3(
      ganache.provider({ fork: "wss://eth-mainnet.ws.alchemyapi.io/v2/MYKEY", fork_block_number: latest_block })
    )

    localWeb3.extend({
      methods: [{
          name: 'unlockAccount',
          call: 'evm_unlockUnknownAccount',
          params: 1
      }]
    })

    console.log('forked web3: ', localWeb3) // object
    console.log("latest_block forked without await: ",localWeb3.eth.getBlockNumber()) // pending
    console.log("latest_block forked with await: ", await localWeb3.eth.getBlockNumber()) // blank
    console.log("unlocked? ", await localWeb3.unlockAccount("SOMEADDRESS")) // blank
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

export default App;
```

Extra Notes:
1. This issue only happen when using ganache.provider(), using `new Web3.providers.WebsocketProvider` connected to `ganache-cli` fork works fine.
2. The same web3 script works without reactjs

**Comment by davidmurdoch** [comment]
Are you running this in a browser?

**Comment by jmarrr** [comment]
> Are you running this in a browser?

Yes

**Comment by davidmurdoch** [comment]
This is likely because ganache-core isn't currently supported in the browser.

I have released a version of ganache _for internal testing_ that does work in the browser. If you want to try it you can install it by running `npm uninstall ganache-core && npm install --exact ganache@internal` or `yarn remove ganache-core && yard add --exact ganache@internal`.

**However, forking is not yet supported in this version.**

There are also many breaking changes in this experimental version and I do not yet have an upgrade guide. To align with `ganache-core@v2` mining behavior, which is the most obvious breaking change, you should enable the new `miner.legacyInstamine` option:

```javascript
const ganache = require("ganache");
const provider = ganache.provider({ miner:{ legacyInstamine: true } });
```

**Comment by gnidan** [comment]
This should already be fixed as part of the upcoming Ganache v7.0.0 release.

Leaving this issue open so we can validate it upon that release.

Thanks!

**Comment by davidmurdoch** [comment]
@jmarrr I'm going to close this one, as ganache v7 (canary release) works in the browser now. You can find an example of this in our (WIP) interactive docs: https://trufflesuite.github.io/ganache-core/
